pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -344,17 +334,32 @@ func (rd *restoreDataProcessor) openSSTs(
 			return err
 		}
 		dirs = append(dirs, dir)
+		filePaths = append(filePaths, file.Path)
 
 		// TODO(pbardea): When memory monitoring is added, send the currently
 		// accumulated iterators on the channel if we run into memory pressure.
-		iter, err := storageccl.ExternalSSTReader(ctx, dir, file.Path, rd.spec.Encryption)
+	}
+	iterOpts := storage.IterOptions{
+		RangeKeyMaskingBelow: rd.spec.RestoreTime,
+		KeyTypes:             storage.IterKeyTypePointsAndRanges,
+		LowerBound:           keys.LocalMax,
+		UpperBound:           keys.MaxKey,
+	}
+
+	if !storageccl.RemoteSSTs.Get(&rd.EvalCtx.Settings.SV) {
```

Comment:
I feel like I sorta preferred just calling one function and inside that function we decide if we should pass the item a seeking reader or a buffer-backed reader, the way ExternalSSTReader did?

Reply:
How's my latest push instead?  My [original implementation](https://github.com/cockroachdb/cockroach/commit/23468391f248c07515af91fbe738d55226fdbc61) kept it all in one func, and it seemed quite messy.

Reply:
David approved over meeting.